src/biogeophys/SurfaceWaterMod.F90

```diff
@@ -472,15 +472,25 @@ subroutine QflxH2osfcSurf(bounds, num_hydrologyc, filter_hydrologyc, &
        c = filter_hydrologyc(fc)
 
        if (h2osfcflag==1) then
-          if (frac_h2osfc_nosnow(c) <= params_inst%pc) then
-             frac_infclust=0.0_r8
-          else
-             frac_infclust=(frac_h2osfc_nosnow(c)-params_inst%pc)**params_inst%mu
-          endif
+	   ! for rice we set the runoff at 0
+         if (col%itype(c)==(200+cft_lb+46) .or. col%itype(c) == (200+cft_lb+47)) then
+             frac_infclust=0.0_r8																			  			 
+         else if (frac_h2osfc_nosnow(c) <= params_inst%pc) then
+            frac_infclust=0.0_r8
+         else
+            frac_infclust=(frac_h2osfc_nosnow(c)-params_inst%pc)**params_inst%mu
+         endif
        endif
 
        ! limit runoff to value of storage above S(pc)
-       if(h2osfc(c) > h2osfc_thresh(c) .and. h2osfcflag/=0) then
+	   ! if surface water exceeds 10cm, then do drainage in the next step
+       if (col%itype(c)==(200+cft_lb+46) .or. col%itype(c)==(200+cft_lb+47)) then
```

Comment:
This will need to use the standard CLM irrigated rice types, `nrice` and `nirrig_rice`. (Although why did you include rainfed rice?)

Reply:
Hey Sam, this one is a bit complicated. I guess I would need to use col%patch(bounds%begp) as there is only one patch for each irrigated crop column right? Will try and test it. Based on my experience, farmers normally change micro topography to hold the water in the field, so I also did it for rainfed rice. I would be very surprised if they don't do it for rainfed rice (check fig1: https://link.springer.com/article/10.1007/s13593-021-00710-2)

Reply:
Ah okay, that makes sense about the microtopography. In that case, it'd be helpful to add a comment explaining your reasoning so that future developers don't get confused like I did.